src/Refinement/RefinementTransferOperators.jl

```diff
@@ -0,0 +1,166 @@
+
+"""
+  RefinementTransferOperator
+
+  Matrix-like operator providing a transfer between two `FESpaces`
+  defined in child-parent refined grids. 
+
+  The dofs are transfered by iteratively solving a projection problem 
+
+        (uh_to,vh_to)_to = (uh_from,vh_to)_Ω
+
+  where Ω is always chosen as the finest grid (i.e the RHS is
+  always integrated without loss of accuracy). 
+"""
+struct RefinementTransferOperator{T,A,B,C} <: AbstractMatrix{T}
+  from   ::A
+  to     ::B
+  caches ::C
+
+  function RefinementTransferOperator(T,from,to,caches)
+    A = typeof(from)
+    B = typeof(to)
+    C = typeof(caches)
+    new{T,A,B,C}(from,to,caches)
+  end
+end
+
+# L2 projection (default for the transfer ops)
+Π_l2(u,v,dΩ) = ∫(v⋅u)*dΩ
+
+function RefinementTransferOperator(from::FESpace,to::FESpace; Π=Π_l2, qdegree=3)
+  @assert isa(from,TrialFESpace)
+  @assert isa(to,TrialFESpace)
+
+  Ω_from = get_triangulation(from)
+  Ω_to   = get_triangulation(to)
+  @assert isa(Ω_from,RefinedTriangulation) || isa(Ω_to,RefinedTriangulation)
+  @assert is_change_possible(Ω_from,Ω_to)
+
+  # Choose integration space
+  Ω  = best_target(Ω_from,Ω_to)
+  dΩ = Measure(Ω,qdegree)
+  U  = (Ω === Ω_from) ? from : to
+  V  = U.space
+  vh_to = get_fe_basis(to)
+  vh = change_domain(vh_to,Ω)
+
+  # Prepare system
+  sysmat, sysvec = assemble_lhs(Π,Ω_to,to,to.space,qdegree)
+  assem  = SparseMatrixAssembler(to,to.space)
+
+  cache = sysmat, sysvec, Π, assem, Ω, dΩ, U, V, vh
+  return RefinementTransferOperator(eltype(sysmat),from,to,cache)
+end
+
+# Solves the problem Π(uh,vh)_to = Π(uh_from,vh)_Ω for all vh in Vh_to
+function LinearAlgebra.mul!(y,A::RefinementTransferOperator,x)
+  sysmat, sysvec, Π, assem, Ω, dΩ, U, V , vh_Ω = A.caches
+  Ω_to = get_triangulation(A.to)
+
+  # Bring uh to the integration domain
+  uh_from = FEFunction(A.from,x)
+  uh_Ω    = change_domain(uh_from,Ω)
+
+  # Assemble rhs vector
+  contr   = Π(uh_Ω,vh_Ω,dΩ)
+  if Ω !== Ω_to
+    contr = merge_contr_cells(contr,Ω,Ω_to)
+  end
+  vecdata = collect_cell_vector(A.to.space,contr)
+  assemble_vector_add!(sysvec,assem,vecdata)
+
+  # Solve projection
+  IterativeSolvers.cg!(y,sysmat,sysvec)
+  return y
+end
+
+function Base.size(A::RefinementTransferOperator)
+  (num_free_dofs(A.to),num_free_dofs(A.from))
+end
+
+function Base.size(A::RefinementTransferOperator,i::Int)
+  if i == 1
+    return num_free_dofs(A.to)
+  elseif i == 2
+    return num_free_dofs(A.from)
+  else
+    return nothing
+  end
+end
+
+function Base.display(op::RefinementTransferOperator{T,A,B,C}) where {T,A,B,C}
+  s = size(op)
+  println("$(s[1])x$(s[2])  RefinementTransferOperator{$(T)}")
+end
+
+function assemble_lhs(Π,Ω,Uh,Vh,qdegree)
+  dΩ = Measure(Ω,qdegree)
+  uh_dir = FEFunction(Uh,zero_free_values(Uh),get_dirichlet_dof_values(Uh))
+  a(u,v) = Π(u,v,dΩ)
+  b(v)   = a(uh_dir,v)
+
+  sysmat = assemble_matrix(a,Uh,Vh)
+  sysvec = assemble_vector(b,Vh)
+  return sysmat, -sysvec
```

Comment:
Why the `-` sign here? 

Why `assemble_lhs` returns a vector and a matrix? Shouldn't be `lhs` (left-hand side) just a matrix?

Reply:
Be careful with `-sysvec`, it creates temporary memory allocation. I would better use `sysvec .= -.sysvec` to avoid this.

Reply:
So from what I understand here `sysvec` is the contributions of the Dirichlet BCs, right? So they come from the LHS (the bilinear form) but end up being a vector constibution that changes sign and goes to the RHS. So that's why it is calculated here and also why it changes sign.

Reply:
I implemented some changes in [caeb975](https://github.com/gridap/Gridap.jl/pull/838/commits/caeb9750be59e93582cfb5444f1bcd47c83001bf) that might make things clearer. Hopefully it is now more obvious what is going on.